pkg/runtime/conversion_generator.go

```diff
@@ -19,40 +19,69 @@ package runtime
 import (
 	"fmt"
 	"io"
+	"path"
 	"reflect"
 	"sort"
 	"strings"
 
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/conversion"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
 )
 
 type ConversionGenerator interface {
 	GenerateConversionsForType(version string, reflection reflect.Type) error
 	WriteConversionFunctions(w io.Writer) error
-	RegisterConversionFunctions(w io.Writer) error
+	RegisterConversionFunctions(w io.Writer, pkg string) error
+	AddImport(pkg string) string
+	RepackImports(exclude util.StringSet)
+	WriteImports(w io.Writer) error
 	OverwritePackage(pkg, overwrite string)
+	AssumePrivateConversions()
```

Comment:
What is the purpose of it? I don't see it used anywhere and I'm not sure if I understand why it is really needed.